xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/upload.js

```diff
@@ -121,12 +121,28 @@ var XWiki = (function(XWiki) {
       var statusUI = this.statusUI = {};
 
       statusUI.UPLOAD_STATUS = UploadUtils.createDiv('upload-status upload-inprogress');
-
+      
+      // Set up the icons and their text alternatives
+      // The inprogress icon is a special case, we set the content using a GIF background in CSS
+      statusUI.STATUS_ICON_INPROGRESS = UploadUtils.createDiv('status-icon icon-inprogress', '');
+      statusUI.STATUS_ICON_INPROGRESS_ALTERNATIVE = UploadUtils.createSpan('sr-only', "$!escapetool.javascript($services.localization.render('core.widgets.html5upload.status.icon.inprogress'))");
+      statusUI.STATUS_ICON_INPROGRESS.insert(statusUI.STATUS_ICON_INPROGRESS_ALTERNATIVE);
+      statusUI.STATUS_ICON_DONE = UploadUtils.createDiv('status-icon icon-done', "$!escapetool.javascript($services.icon.renderHTML('check'))");
+      statusUI.STATUS_ICON_DONE_ALTERNATIVE = UploadUtils.createSpan('sr-only', "$!escapetool.javascript($services.localization.render('core.widgets.html5upload.status.icon.done'))");
+      statusUI.STATUS_ICON_DONE.insert(statusUI.STATUS_ICON_DONE_ALTERNATIVE);
+      statusUI.STATUS_ICON_CANCELED = UploadUtils.createDiv('status-icon icon-canceled', "$!escapetool.javascript($services.icon.renderHTML('remove'))");
+      statusUI.STATUS_ICON_CANCELED_ALTERNATIVE = UploadUtils.createSpan('sr-only', "$!escapetool.javascript($services.localization.render('core.widgets.html5upload.status.icon.canceled'))");
+      statusUI.STATUS_ICON_CANCELED.insert(statusUI.STATUS_ICON_CANCELED_ALTERNATIVE);
+      statusUI.STATUS_ICON_ERROR = UploadUtils.createDiv('status-icon icon-error', "$!escapetool.javascript($services.icon.renderHTML('error'))");
+      statusUI.STATUS_ICON_ERROR_ALTERNATIVE = UploadUtils.createSpan('sr-only', "$!escapetool.javascript($services.localization.render('core.widgets.html5upload.status.icon.error'))");
+      statusUI.STATUS_ICON_ERROR.insert(statusUI.STATUS_ICON_ERROR_ALTERNATIVE);
+      statusUI.UPLOAD_STATUS.insert(statusUI.STATUS_ICON_INPROGRESS).insert(statusUI.STATUS_ICON_DONE).insert(statusUI.STATUS_ICON_CANCELED).insert(statusUI.STATUS_ICON_ERROR);
+      
```

Comment:
Not great to mix velocity and javascript for translations. You should use the bestpractice documented in https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HJavaScriptBestPractices

Reply:
So, I tried applying the best practice, but this is some Prototype.js and it broke things. 
Instead, I decided to group up all of these translations at the start as seen on https://github.com/xwiki/xwiki-platform/blob/6bbb277528374f801a3c256fd281234db844d9bb/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/importer/import.js#L24C1-L43 . See https://github.com/xwiki/xwiki-platform/pull/2700/commits/6af9113d21e1d16df648c8c06223be129326ef3e

Note that most of those translations were already there before the PR, they are changed to make escaping consistent in this file after https://github.com/xwiki/xwiki-platform/pull/2700#discussion_r1419340603